kanidm_proto/src/v1.rs

```diff
@@ -98,6 +98,92 @@ pub enum PasswordFeedback {
+                write!(f, "Capitalization doesn't help very much.")
+            }
+            PasswordFeedback::CommonNamesAndSurnamesAreEasyToGuess => {
+                write!(f, "Common names and surnames are easy to guess.")
+            }
+            PasswordFeedback::DatesAreOftenEasyToGuess => {
+                write!(f, "Dates are often easy to guess.")
+            }
+            PasswordFeedback::NamesAndSurnamesByThemselvesAreEasyToGuess => {
+                write!(f, "Names and surnames by themselves are easy to guess.")
+            }
+            PasswordFeedback::NoNeedForSymbolsDigitsOrUppercaseLetters => {
+                write!(f, "No need for symbols, digits or upper-case letters.")
+            }
+            PasswordFeedback::PredictableSubstitutionsDontHelpVeryMuch => {
+                write!(f, "Predictable substitutions don't help very much.")
+            }
+            PasswordFeedback::RecentYearsAreEasyToGuess => {
+                write!(f, "Recent years are easy to guess.")
+            }
+            PasswordFeedback::RepeatsLikeAaaAreEasyToGuess => {
+                write!(f, "Repeats like 'aaa' are easy to guess.")
+            }
+            PasswordFeedback::RepeatsLikeAbcAbcAreOnlySlightlyHarderToGuess => write!(
+                f,
+                "Repeats like abcabcabc are only slightly harder to guess."
+            ),
+            PasswordFeedback::ReversedWordsArentMuchHarderToGuess => {
+                write!(f, "Reversed words aren't much harder to guess.")
+            }
+            PasswordFeedback::SequencesLikeAbcAreEasyToGuess => {
+                write!(f, "Sequences like 'abc' are easy to guess.")
+            }
+            PasswordFeedback::ShortKeyboardPatternsAreEasyToGuess => {
+                write!(f, "Short keyboard patterns are easy to guess.")
+            }
+            PasswordFeedback::StraightRowsOfKeysAreEasyToGuess => {
+                write!(f, "Straight rows of keys are easy to guess.")
+            }
+            PasswordFeedback::ThisIsACommonPassword => write!(f, "This is a common password."),
+            PasswordFeedback::ThisIsATop100Password => write!(f, "This is a top 100 password."),
+            PasswordFeedback::ThisIsATop10Password => write!(f, "This is a top 10 password."),
+            PasswordFeedback::ThisIsSimilarToACommonlyUsedPassword => {
+                write!(f, "This is similar to a commonly used password.")
+            }
+            PasswordFeedback::TooShort(minlength) => write!(
+                f,
+                "Password too was short, needs to be at least {} characters long.",
+                minlength
+            ),
+            PasswordFeedback::UseAFewWordsAvoidCommonPhrases => {
+                write!(f, "Use a few words and avoid common phrases.")
+            }
+            PasswordFeedback::UseALongerKeyboardPatternWithMoreTurns => {
+                write!(f, "Use a longer keyboard pattern with more turns.")
```

Comment:
What is a "turn" ?

Reply:
I... don't know... but [this thread](https://github.com/dropbox/zxcvbn/issues/196) indicates it's about  "don't just run your hand across the keyboard in one direction". It's a complex one, so I updated the text a little.

Reply:
Thanks!